Neuland new calibration

P-Ulrich opened the pull request:
Describe your proposal.

Mention any issue this PR is resolved or is related to.

---

Checklist:

* [x] Rebased against `dev` branch
* [x] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [x] Followed the [pull request guidlines](https://opensource.creativecommons.org/contributing-code/pr-guidelines/) and the [Git workflow](https://github.com/AnarManafov/GitWorkflow/blob/master/GitWorkflow.markdown)
* [x] Followed the [seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

YanzhaoW committed:
neuland calibration process reformat and millepde

P-Ulrich committed:
Implementation of neuland moun simulation.

YanzhaoW committed:
add neuland simCal to CalData converter and others

- fix the rebase for origin dev
- create neuland_ana with JSON config
- resolve issues from merge
- fix r3bfield error
- add neuland geo creator
- set common options in app base class
- complete sim json exe
- finsih the program options
- automate image build and push
- add simcal2cal
- fix the compile after rebase
- fix the ci issue from doxygen

P-Ulrich committed:
corrected reverseatten and gen sphere koord change

YanzhaoW commented on neuland/simulation/generator/R3BNeulandGenerators.h:
```diff
@@ -13,7 +13,7 @@ inline auto create_muon_generator(TRandom3& random_gen)
     constexpr auto num_DP = 13;
     constexpr auto PID = 2112;
-    constexpr auto z_pos = 0.;
+    constexpr auto z_pos = 1650.0;
```
This value should not be hard coded in the source code. Better make it as a member variable and set it using a setter public member function.

YanzhaoW committed:
delete neulandAna and neulandSim and add doxygen auto-build

P-Ulrich committed:
more corrected atten

fixed attenuation and add more documentation to exe/readme

- update R3BNeuland.h documentation
- change back to macOS doxygen
- add magic_enum
- fix doxygen readme.md problem
- Set conan cppstd to gnu17
- Use CMAKE_CXX_STANDARD to be 17
- Add CONAN_ENABLED guards
- fix rebase problem

add init simcal

YanzhaoW committed:
Resolve CI issues

- Disable CI for sub-repos and debian 10
- add PEDEDIR definition in CI
- add source config.sh to ctest
- cmakelists checked with dev branch
- Remove the AddVolume from gGeoManager
- This eliminate the duplication of volNeuland volume
- Finish the transition to R3BIOConnector
- Transform the neuland reconstruction algorithm to R3BIOVector

P-Ulrich committed:
add parameter neulandpaddle

YanzhaoW committed:
change github ci to edwin_dev trigger
